src/tests/ftest/tags.py

```diff
@@ -650,90 +747,211 @@ def print_verbose(*args):
+                if tags and not tag_map.is_test_subset([method_tags], tags):
+                    continue
+
+                # Add a new row of output
+                try:
+                    output.append(
+                        [frequency_id(method_tags), short_file_path, class_name, method_name])
+                    yaml_data = TestConfig(file_path, method_name)
+                    for key in key_list:
+                        output[-1].append(yaml_data.value(key))
+                except Exception as error:      # pylint: disable=broad-except
+                    print(f"<< Error processing yaml keys {key_list} for {short_file_path} >>")
+                    raise error
+
+    if csv_file:
+        with open(csv_file, 'w', newline='\n') as csvfile:
+            csv_writer = csv.writer(csvfile)
+            for line in output:
+                csv_writer.writerow(line)
+        print(f"Generated {csv_file} with {len(output)} records")
+
+    else:
+        widths = [0] * len(output[0])
+        for line in output:
+            for index, column in enumerate(line):
+                widths[index] = max(widths[index], len(str(column)))
+        format_line = "  ".join(list(map(lambda x: f"{{:{x}}}", widths)))
+        for line in output:
+            print(format_line.format(*list(map(str, line))))
+
+    return 0 if output else 1
+
+
+def __comma_separated_arg(val):
+    """Parse a comma-separated argument.
+
+    Args:
+        val (str): string to parse comma-separated values from
+
+    Returns:
+        set: comma-separated strings converted to a set
 
     Raises:
         ArgumentTypeError: if val is invalid
     """
     if not val:
-        raise ArgumentTypeError("tags cannot be empty")
+        raise ArgumentTypeError("comma-separated argument cannot be empty")
     try:
         return set(map(str.strip, val.split(",")))
     except Exception as err:  # pylint: disable=broad-except
-        raise ArgumentTypeError(f"Invalid tags: {val}") from err
+        raise ArgumentTypeError(f"Invalid comma-separated argument: {val}") from err
 
 
 def main():
-    """main function execution"""
-    description = '\n'.join([
-        'Commands',
-        '  lint - lint ftest avocado tags',
-        '  list - list ftest avocado tags associated with test files',
-        '  dump - dump the file/class/method/tag structure for test files',
-        '  unit - run self unit tests'
-    ])
-
-    parser = ArgumentParser(formatter_class=RawDescriptionHelpFormatter, description=description)
-    parser.add_argument(
-        "command",
-        choices=("lint", "list", "dump", "unit"),
-        help="command to run")
-    parser.add_argument(
+    """Main function execution.
+
+    Returns:
+        int: 0 = success
+             1 = error
+    """
+    parser = ArgumentParser(prog='tags')
+    subparsers = parser.add_subparsers(
+        title='options for the tags command',
+        dest='command')
```

Comment:
We need `required=True` since it is required

Reply:
It does not appear so:
```
$ ./tags.py
usage: tags.py [-h] [-v] [--paths PATHS [PATHS ...]] [--tags TAGS [TAGS ...]] {lint,list,dump,unit}
tags.py: error: the following arguments are required: command

Reply:
Yes, thanks.